### DISCUSSION TITLE
How to do conditional vite config depending on mode (i.e. dev or prod)?

### DISCUSSION THREAD

**Discussion by lominming** [solidjs/solid-start#1290] [Q&A]
In pre-beta2, we can still use this https://vitejs.dev/config/#conditional-config where `defineConfig` can accept a function that has `mode` that determine development or production. 

However in beta-2, we can no longer do it. Seems like an obvious enough requirement to change certain configurations depending on dev or prod build. How can I still do that in beta2?

Couple of examples in my config file:
1) `import { loadEnv } from 'vite'` and `loadEnv(mode, ...)` function requires to pass in the mode `development` or `production`
2) dropping console.log if `mode === 'production'`

**Comment by atk** [comment]
That is because we are no longer directly using vite, but we are using vinxi to set vite up. Our configuration is no longer a single object containing a single server, it is containing "routes" that lead to different server configurations that can deliver static data, client bundles, server rendered pages and user middlewares by default.

This is currently being discussed in https://github.com/solidjs/solid-start/discussions/1279#discussioncomment-8228258 - but there's no eta on when this is being fixed yet.

**Reply by TheKnightCoder** [reply]
Was this fixed?
